upgrade/steps-helpers.go

```diff
@@ -506,33 +505,26 @@ func getExpectedTargetFromIssues(ctx Context, name string) (*UpstreamUpgradeTarg
 			continue
 		}
 		v, err := semver.NewVersion(version)
-		if err == nil {
-			if ctx.InferVersion && !(ctx.TargetVersion == nil || ctx.TargetVersion.Equal(v) || ctx.TargetVersion.GreaterThan(v)) {
-				versionConstrained = true
-				continue
-			}
-			versions = append(versions, UpgradeTargetIssue{
-				Version: v,
-				Number:  title.Number,
-			})
+		if err != nil {
+			continue
 		}
+		versions = append(versions, UpgradeTargetIssue{
+			Version: v,
+			Number:  title.Number,
+		})
 	}
 	if len(versions) == 0 {
-		var extra string
-		if ctx.InferVersion && versionConstrained {
-			extra = " (a version was found but it was greater then the specified max)"
-		}
-		return nil, extra, nil
+		return nil, nil
 	}
 	sort.Slice(versions, func(i, j int) bool {
 		return versions[j].Version.LessThan(versions[i].Version)
 	})
 
-	if ctx.InferVersion && ctx.TargetVersion != nil && !versions[0].Version.Equal(ctx.TargetVersion) {
-		return nil, "", fmt.Errorf("possible upgrades exist, but non match %s", ctx.TargetVersion)
+	if ctx.TargetVersion != nil && !versions[0].Version.Equal(ctx.TargetVersion) {
+		return nil, fmt.Errorf("possible upgrades exist, but non match %s", ctx.TargetVersion)
```

Comment:
`s/non/none/` 

Also I'm unclear why we're only testing against `versions[0]` rather than searching `versions` for a match?

Reply:
We should be doing that. Fixed in https://github.com/pulumi/upgrade-provider/pull/136/commits/2d2e40c4b4cc0e849ee639331e95f22018ccbc56.